docs/production/authentication-methods.md

```diff
@@ -669,6 +669,12 @@ domain for your server).
 [apple-get-started]: https://developer.apple.com/sign-in-with-apple/get-started/
 [outgoing-email]: ../production/email.md
 
+## Generic OpenID Connect
+
+Zulip can be integrated with any OpenID Connect authentication provider.
```

Comment:
Maybe do "OpenID Connect (OIDC)" in case folks `Ctrl+F` search for it.

We should also do a bit of a `git grep SAML` audit and mention this as a possibility where we talk about supported authentication backends.